llvm/include/llvm/Object/DXContainer.h

```diff
@@ -117,14 +117,24 @@ template <typename T> struct ViewArray {
 
 namespace DirectX {
 
+struct RootParameter {
+  dxbc::RootParameterHeader Header;
+  union {
+    dxbc::RootConstants Constants;
+  };
+};
 class RootSignature {
 private:
-  uint32_t Version;
-  uint32_t NumParameters;
-  uint32_t RootParametersOffset;
-  uint32_t NumStaticSamplers;
-  uint32_t StaticSamplersOffset;
-  uint32_t Flags;
+  uint32_t Version = 2;
+  uint32_t NumParameters = 0;
+  uint32_t RootParametersOffset = 0;
+  uint32_t NumStaticSamplers = 0;
+  uint32_t StaticSamplersOffset = 0;
+  uint32_t Flags = 0;
+
+  SmallVector<DirectX::RootParameter> Parameters;
+
+  using ParamsIter = SmallVector<DirectX::RootParameter>::iterator;
```

Comment:
LLVM generally uses a naming convention like `foo_iterator` for things like this (see `Value::use_iterator` or `Function::param_iterator`) to match the STL naming conventions.

We should call this type `param_iterator`.